lms/static/js/learner_dashboard/models/course_enroll_model.js

```diff
@@ -0,0 +1,20 @@
+/**
+ *  Track optIn status and submit form
+ */
+;(function (define) {
+    'use strict';
+
+    define([
+            'backbone'
+        ],
+        function( Backbone) {
+            return Backbone.Model.extend({
+                defaults: {
+                    course_id: '',
+                    optIn: false,
+                },
+                urlRoot: '/api/commerce/v0/baskets/'
```

Comment:
I think it's due to the fact we worry about the dependency on the backend data. If this URL is being passed only from backend, what if that url is non-existent? Should we render this view? Or should we create a default so we at least have somewhere to post to?
I think the solution I am leaning towards is that we leave the hardcoded url in, but also accept an override from backend for the enrollment url. Does that make sense?